interfaces/cython/cantera/composite.py

```diff
@@ -565,6 +566,30 @@ def equilibrate(self, *args, **kwargs):
             self._phase.equilibrate(*args, **kwargs)
             self._states[index][:] = self._phase.state
 
+    def set_equivalence_ratio(self, phi, *args, **kwargs):
+        """
+        See `ThermoPhase.set_equivalence_ratio` 
+
+        Note that *phi* either needs to be a scalar value or dimensions have
+        to be matched to the SolutionArray.
+        """
+
+        # test arguments
+        if isinstance(phi, _numbers.Real):
+            phi = phi * np.ones(self._shape)
+        elif isinstance(phi, np.ndarray):
+            if phi.shape!=self._shape:
+                raise ValueError('Shape mismatch.')
+        else:
+            if len(phi)!=len(self._indices):
+                raise ValueError('Incompatible dimensions.')
```

Comment:
The alternative to this block is to use the same approach as is used in the state setting functions:
```python
    phi, _ = np.broadcast_arrays(phi, self._output_dummy)
```
This also allows things like broadcasting a `phi` array to a higher-dimensional `SolutionArray`, e.g. from 1xN to MxN.

Reply:
NP. I was looking for something existing but obviously missed that one ...